lib/Conversion/HandshakeToHW/HandshakeToHW.cpp

```diff
@@ -776,6 +777,8 @@ class JoinConversionPattern : public HandshakeConversionPattern<JoinOp> {
                    hw::HWModulePortAccessor &ports) const override {
     auto unwrappedIO = unwrapIO(s, bb, ports);
     buildJoinLogic(s, unwrappedIO.inputs, unwrappedIO.outputs[0]);
+    unwrappedIO.outputs[0].data->setValue(
```

Comment:
As we do not write the value in `buildJoinLogic`, we shouldn't do it either for `buildForkLogic`. 
It's clear for a join that input's might have different types, and the same can hold or operations using fork logic.

Reply:
Alternatively, we could add a `isControl` / `controlOnly` flag that would optionally drop the assignments of the data ports. "control" might be a bit misleading, though.

Reply:
I'd be fine with leaving the data assignment out of `buildForkLogic` and let the caller handle how data is assigned. will change.